wallet-new/src/Cardano/Wallet/Kernel/DB/AcidState.hs

```diff
@@ -104,17 +105,26 @@ newPending accountId tx = runUpdate' . zoom dbHdWallets $
     zoom hdAccountCheckpoints $
       mapUpdateErrors NewPendingFailed $ Spec.newPending tx
 
--- | Cancels the input transactions from all the 'Checkpoints' of each of
--- the accounts cointained in all the wallets.
+-- | Errors thrown by 'cancelPending'
+data CancelPendingError =
+    -- | Unknown account
+    CancelPendingUnknown UnknownHdAccount
+
+deriveSafeCopy 1 'base ''CancelPendingError
+
+-- | Cancels the input transactions from the 'Checkpoints' of each of
+-- the accounts cointained in the 'Cancelled' map.
 --
 -- The reason why this function doesn't take a 'HdRootId' as an argument
 -- is because the submission layer doesn't have the notion of \"which HdWallet
 -- is this transaction associated with?\", but it merely dispatch and cancels
 -- transactions for all the wallets managed by this edge node.
-cancelPending :: InDb (Set Core.TxId) -> Update DB ()
-cancelPending tx =
-    runUpdateNoErrors . zoomAll (dbHdWallets . hdWalletsAccounts) $
-        hdAccountCheckpoints %~ Spec.cancelPending tx
+cancelPending :: InDb Cancelled -> Update DB (Either CancelPendingError ())
+cancelPending (InDb cancelled) = runUpdate' . zoom dbHdWallets $
+    forM_ (Map.toList cancelled) $ \(accountId, txids) -> do
```

Comment:
Here is another bug -- due to the short-circuiting nature of `ExceptT`, if one of the zoom would fail due to an missing account, the other txs won't be evicted as we would stop sooner. Nonsense!